Assets/Scripts/MaquinaDeRegras.cs

```diff
@@ -24,232 +25,287 @@ private int[] SetValoresPecas(int jogador)
+        return melhor;
     }
 
-    private List<List<int>> HighlightDamas(int[][] tabuleiro, int x, int y, int pecaInimiga, int damaInimiga)
+    // retorna os possíveis movimentos para dama
+    private List<Jogada> MovimentosDama(int[,] tabuleiro, int x, int y, int pecaInimiga, int damaInimiga)
     {
-        List<List<int>> posicoesCimaDireita = new List<List<int>>();
-        List<List<int>> posicoesCimaEsquerda = new List<List<int>>();
-        List<List<int>> posicoesBaixoDireita = new List<List<int>>();
-        List<List<int>> posicoesBaixoEsquerda = new List<List<int>>();
+        Jogada jogadaCimaDireita = new Jogada();
+        Jogada jogadaCimaEsquerda = new Jogada();
+        Jogada jogadaBaixoDireita = new Jogada();
+        Jogada jogadaBaixoEsquerda = new Jogada();
         int i, j;
-        bool[] controle = {false, true}; // comer, podeMover
-        for(i = x, j = y; i < 8 && j < 8; i++, j++){
-            controle = checkFor(posicoesCimaDireita, tabuleiro, i, j, pecaInimiga, damaInimiga, controle[0]);
-            if(!controle[1]) // não há movimentação nessa diagonal
-                break;
-        }
-        for(i = x, j = y; i < 8 && j >= 0; i++, j--){
-            controle = checkFor(posicoesBaixoDireita, tabuleiro, i, j, pecaInimiga, damaInimiga, controle[0]);
-            if(!controle[1]) // não há movimentação nessa diagonal
-                break;
-        }
-        for(i = x, j = y; i >= 0 && j < 8; i--, j++){
-            controle = checkFor(posicoesCimaEsquerda, tabuleiro, i, j, pecaInimiga, damaInimiga, controle[0]);
-            if(!controle[1]) // não há movimentação nessa diagonal
-                break;
-        }
-        for(i = x, j = y; i >= 0 && j >= 0; i--, j--){
-            controle = checkFor(posicoesBaixoEsquerda, tabuleiro, i, j, pecaInimiga, damaInimiga, controle[0]);
-            if(!controle[1]) // não há movimentação nessa diagonal
-                break;
-        }
-        if (controle[0]) // comer == true?
-            return max(posicoesBaixoDireita, posicoesBaixoEsquerda, posicoesCimaDireita, posicoesCimaEsquerda);
-        else
+        // TODO chamar LeiDaMaioria aqui
+        for(i = x+1, j = y+1; i < 8 && j < 8; i++, j++)
         {
-            List<List<int>> posicoesTotais = new List<List<int>>();
-            foreach (List<int> posicao in posicoesBaixoDireita)
-                posicoesTotais.Add(posicao);
-            foreach (List<int> posicao in posicoesBaixoEsquerda)
-                posicoesTotais.Add(posicao);
-            foreach (List<int> posicao in posicoesCimaDireita)
-                posicoesTotais.Add(posicao);
-            foreach (List<int> posicao in posicoesCimaEsquerda)
-                posicoesTotais.Add(posicao);
-            return posicoesTotais;
-        }
-    }
+            if(Tipos.isVazio(tabuleiro[i,j]))
+            {
+                jogadaCimaDireita.movimentos.Add(new int[2] {i,j});
+                // chamar LeiDaMaioria aqui
```

Comment:
adiciona a tag TODO plz pra gente conseguir pesquisar dps

Reply:
isso não é bem um TODO, é mais algo para manter meu raciocinio quando for montar o restante dessa função. de qualquer forma tem uma issue para lidar com a representação disso

Reply:
ok